src/aria/popups/PopupNavigationManager.js

```diff
@@ -433,5 +642,236 @@ function ViewportNavigationInterceptor() {
+                var origin = navigationInformation.origin;
+
+                var reverse = (direction === 'forward' && origin === 'popup') || (direction === 'backward' && origin === 'browser');
+                if (loop) {
+                    reverse = !reverse;
+                }
+                ariaTemplatesNavigationManager.focusFirst(element, reverse);
+            }
+        }
+    };
+
+    // -------------------------------------------------------------------------
+
+    var elementInterceptor = DialogNavigationInterceptor({
+        onfocus: onfocus,
+        getReferenceElement: function() {return element;}
+    });
+
+    var viewportInterceptor = ViewportNavigationInterceptor({
+        onfocus: onfocus
+    });
+
+    var compositeInterceptor = CompositeInterceptor([
+        elementInterceptor,
+        viewportInterceptor
+    ]);
+
+    // ------------------------------------------------------------------ return
+
+    return compositeInterceptor;
+}
+exports.ElementNavigationInterceptor = ElementNavigationInterceptor;
+
+
+
+////////////////////////////////////////////////////////////////////////////////
+// Hiding manager
+////////////////////////////////////////////////////////////////////////////////
+
+/**
+ * Manages the hiding/showing of elements through the WAI-ARIA attribute "aria-hidden".
+ *
+ * <p>
+ * It manages hidden elements by tracking the number of request made to hide them or show them back. An element will be hidden the first time one asks, and will be showed back only when no one wants it hidden anymore.
+ * </p>
+ */
+var HidingManager = createConstructor(function() {
+    this.attributeName = 'data-hide-requests-count';
+});
+exports.HidingManager = HidingManager;
+var prototype = HidingManager.prototype;
+
+/**
+ * Hides all elements around the given element, by hiding the nodes as close as possible to the root, avoiding any ancestor of the current node.
+ *
+ * @param {HTMLElement} element The element to keep visible
+ *
+ * @return {Function} A function that can revert the operation done by this method.
+ */
+prototype.hideOthers = function(element) {
+    // -------------------------------------------------------------- processing
+
+    var self = this;
+    var reverters = traverseSiblingsUntilRootOfBranch(element, function(element) {
+        self.hide(element);
+        return function () {
+            self.show(element);
+        };
+    });
+
+    reverters = ariaUtilsArray.flatten(reverters);
+    var reverter = ariaUtilsFunction.bind(
+        ariaUtilsArray.forEach,
+        ariaUtilsArray,
+        reverters,
+        ariaUtilsFunction.call
+    );
+
+    // ------------------------------------------------------------------ return
+
+    return reverter;
+};
+
+/**
+ * Show back all elements around the given element, by applying the same traversal as for hideOthers.
+ *
+ * @param {HTMLElement} element The reference element
+ */
+prototype.showOthers = function(element) {
```

Comment:
Actually this can be used outside of this framework (and I know it is, even though here I preferred the method of the revert function). That brings two different ways to achieve the same thing theoretically.

But I just realized that indeed things start to differ when DOM elements are being added in the meantime, and there is a problem if some code requests hiding those elements. To avoid such things I would have to bind a concept of ids to the requests, thing I wanted to avoid by using a counter instead.